### ISSUE TITLE
 `MODULE_NOT_FOUND` error encountered when attempting to run oclif CLI project command from compiled binary for npm workspaces

### ISSUE THREAD

**Issue by capsulecorplab** [oclif/oclif#1560]
**Is this a bug with the oclif CLI? Or the oclif framework?**
oclif CLI

**Describe the bug**
I have an oclif project that uses npm workspaces to separate logic into their dedicated libraries (e.g., in-memory logic vs filesystem-specific logic). Running a oclif CLI command in dev mode is able to properly call the required libraries from the `packages/` directory. However, running the same command from the compiled binary created from `oclif pack tarballs --targets linux-x64` yields a  `MODULE_NOT_FOUND` error. Below is a screenshot of my oclif CLI app command being run in dev mode versus a compiled binary that was extracted from the gz file created from the `oclif pack tarballs --targets linux-x64` command.

![Screenshot from 2024-09-26 19-45-10](https://github.com/user-attachments/assets/2981c105-1403-4ec3-aa35-666948172479)

**To Reproduce**

1. Clone repo `git clone m30pm https://github.com/Mach30/m30pm.git`
2. `cd` into repo
3. Install dependencies by running `npm install`
4. Build project by running `npm run buildAll`
5. Pack tarballs for linux-x64 architecture by running `oclif pack tarballs --targets linux-x64`
6. `cd` into `dist/` directory
7. Unarchive tarballs with `tar -xvf m30pm-*`
8. `cd` into unarchived `m30pm/bin` directory
9. Run `./m30pm project create my-project`

**Expected behavior**
Should be able to run command without encountering  `MODULE_NOT_FOUND` error

**Your Environment (please complete the following information):**

- `oclif` v4.14.6
- OS & version: Ubuntu 20.04
- Shell/terminal & version 5.0.17

See https://github.com/Mach30/kasm-dof-workspace

**Additional context**
Problem witnessed/captured in https://github.com/Mach30/m30pm/issues/48

**Comment by mdonnalley** [comment]
@capsulecorplab Sorry for taking so long to respond to this and thanks for the detailed writeup.

I'm not exactly sure what the issue might be and I'm not super familiar with using workspaces but I found a couple of things that might be helpful.

If you look at the node_modules, you'll see that the packages are correctly linked:
```
❯ ls -ltra node_modules | grep m30pm
lrwxr-xr-x mdonnalley staff  24 B  Wed Oct  9 16:11:15 2024 m30pm-lib-fs ⇒ ../packages/m30pm-lib-fs
lrwxr-xr-x mdonnalley staff  28 B  Wed Oct  9 16:11:15 2024 m30pm-lib-common ⇒ ../packages/m30pm-lib-common
lrwxr-xr-x mdonnalley staff  25 B  Wed Oct  9 16:11:15 2024 m30pm-lib-cli ⇒ ../packages/m30pm-lib-cli
```

But if you look inside of those, there's no compiled source code:
```
❯ ls packages/m30pm-lib-common
 .   ..   node_modules
```

I thought that maybe changing your `prepack` script to use `npm run buildAll` instead of `npm run build` might have gotten you the compiled source, but that didn't work

I next tried a plain `npm pack` and unpacked the resulting tarball. If you look inside of that, I don't see any of the compiled code for `packages` either - this leads me to believe that it's not an oclif issue since there might be something that you're doing wrong in the tsconfig or in how you configured your workspaces. Again, I'm not very familiar with workspaces so I'm not sure what specifically might be wrong

**Comment by capsulecorplab** [comment]
Getting a similar error when installing and running the oclif CLI app as a debian package, but I can at least print out help text
![Screenshot from 2024-10-25 22-13-32](https://github.com/user-attachments/assets/acc9208c-e294-4e34-b9f3-54778f242e44)

**Comment by capsulecorplab** [comment]
Resolved issue as per https://github.com/Mach30/m30pm/commit/f1094756eb787a94709be90f317214d38ddbef8d; needed to specify `"packages/"` directory under the `"files"` field in `package.json`

**Comment by j-simmons-phd** [comment]
> Resolved issue as per [Mach30/m30pm@f109475](https://github.com/Mach30/m30pm/commit/f1094756eb787a94709be90f317214d38ddbef8d); needed to specify `"packages/"` directory under the `"files"` field in `package.json`

And by following the advice provided to update the `prepack` script to use our `buildAll` command.  Thank you @mdonnalley for getting us pointed in the right direction.
